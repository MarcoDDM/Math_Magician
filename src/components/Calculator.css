.container {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: fit-content;
  margin: 0 auto;
  border: 1px solid black;
  background-color: white;
}

.calculator {
  width: 21rem;
  height: auto;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 1.5rem;
}

.inputContainer {
  width: 19rem;
  height: 3rem;
  text-align: right;
  background-color: rgba(196, 196, 196, 0.89);
  color: rgb(0, 0, 0);
  font-size: 1.5rem;
  padding: 0 1rem;
  margin: 0 0 1rem;
  margin-top: 0;
  margin-bottom: 0.5rem;
  border: 1px solid black;
}

.calculatorButton {
  padding: 1rem 2rem;
  width: 5rem;
  margin: 2px;
  cursor: pointer;
  font-size: 1.1rem;
  border: 1px solid black;
}

.calculatorButton:active {
  background-color: #cfcfcfe2;
}

.mathButton {
  background-color: rgb(199, 199, 199);
}

.mathButton:active {
  background-color: #cfcfcfe2;
}

.mathButtonOrange {
  background-color: rgba(255, 132, 0, 0.92);
}

.mathButtonOrange:active {
  background-color: rgb(255, 156, 50);
}

.zero {
  width: 10.2rem;
}
