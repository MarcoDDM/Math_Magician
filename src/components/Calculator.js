import React, { useState } from 'react';
import './Quotes.css';
import './Calculator.css';
import calculate from '../logic/calculate';
import ChildComponent from './ChildComponent';

const Calculator = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);
  };

  return (
    <div className="container" data-testid="container">
      <ChildComponent
        calcData={calcData}
        handleButtonClick={handleButtonClick}
      />
    </div>
  );
};

export default Calculator;
