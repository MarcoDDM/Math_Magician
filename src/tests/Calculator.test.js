import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Calculator component', () => {
  let calculator;
  beforeEach(() => {
    render(<Calculator />);
    calculator = screen.getByTestId('container');
  });

  it('Should test if Calculate component is inside the document', () => {
    expect(calculator).toBeInTheDocument();
  });

  it('Should render a snapshot of the Calculator component', () => {
    expect(calculator).toMatchSnapshot();
  });
});
