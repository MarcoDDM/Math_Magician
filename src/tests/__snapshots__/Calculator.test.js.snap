// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Calculator component Should render a snapshot of the Calculator component 1`] = `
<div
  class="container"
  data-testid="container"
>
  <div
    class="calculator container"
  >
    <input
      class="inputContainer"
      id="myInput"
      name="name"
      pattern="[0-9+\\\\-*/(). ]+"
      type="text"
      value=""
    />
    <div
      class="buttonContainer"
    >
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        AC
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        +/-
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        %
      </button>
      <button
        class="mathButton calculatorButton operationButton"
        type="button"
      >
        รท
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        7
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        8
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        9
      </button>
      <button
        class="mathButton calculatorButton operationButton"
        type="button"
      >
        x
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        4
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        5
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        6
      </button>
      <button
        class="mathButton calculatorButton operationButton"
        type="button"
      >
        -
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        1
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        2
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        3
      </button>
      <button
        class="mathButton calculatorButton operationButton"
        type="button"
      >
        +
      </button>
      <button
        class="calculatorButton zero mathButton"
        type="button"
      >
        0
      </button>
      <button
        class="calculatorButton mathButton"
        type="button"
      >
        .
      </button>
      <button
        class="mathButton calculatorButton equalsButton"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
